File: EduLink.Domain/Strategies/TransferenciaPagoStrategy.cs
using EduLink.Domain.Interfaces;
using System;
using System.Threading.Tasks;

namespace EduLink.Domain.Strategies;

public class TransferenciaPagoStrategy : IPagoStrategy
{
    public string Nombre => "Transferencia";
    public Task<bool> ProcesarAsync(decimal monto, Guid clienteId)
    {
        Console.WriteLine($"Procesando transferencia por ${monto:F2} a través de Movistar...");
        return Task.FromResult(true);
    }
}

File: app/domain/strategies/transferencia_pago_strategy.rb
require_relative '../interfaces/i_pago_strategy'

module Strategies
  class TransferenciaPagoStrategy
    include Interfaces::IPagoStrategy

    def nombre
      "Transferencia"
    end

    def procesar(monto, cliente_id)
      puts "Procesando transferencia por #{monto}..."
      true # Simula un pago exitoso
    end
  end
end
